src/goto-programs/goto_program.h

```diff
@@ -383,6 +383,13 @@ class goto_programt
     //! This is -1 if it's not a target.
     unsigned target_number;
 
+    //! Id of the scope within which a variable is declared (i.e., DECL).
+    //! It does not have a lot of meaning for other types of instructions.
+    unsigned int scope_id = 0;
```

Comment:
Shall we use BigInt here?

Reply:
I don't think this is necessary since `scope_id` and `parent_scope_id` need to be unique only within a given function (in the current implementation of the algorithm for identifying scopes). I think `unsigned int` is enough to store this information.